### PR TITLE
OKTA-508890 Update the android browser redirect guide to okta-mobile-kotlin.

### PR DESCRIPTION
**Pull request by JayNewstrom**
Related to #3424

Latest Preview Build: 7/8 5PM Eastern
https://62c89674bfb048005cbca58e--reverent-murdock-829d24.netlify.app

## Description:
- **What's changed?** 
   Update https://developer.okta.com/docs/guides/sign-into-mobile-app-redirect/android/main/ from https://github.com/okta/okta-oidc-android to https://github.com/okta/okta-mobile-kotlin in preparation of the okta-mobile-kotlin 1.0 release.
- **Is this PR related to a Monolith release?** 
   No

### Resolves:

* [OKTA-508890](https://oktainc.atlassian.net/browse/OKTA-508890)

### CODE REVIEW

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/createproject.md [hunk 1]
```diff
@@ -1,18 +1,11 @@
...
-
-   ```gradle
-   compileOptions {
-     sourceCompatibility JavaVersion.VERSION_11
-     targetCompatibility JavaVersion.VERSION_11
-   }
-   ```
-
-> **Note**: If you're using the Okta CLI, you can also run `okta start android-java` to create an app. This command creates an OIDC app in Okta, downloads the [okta-android-java-sample](https://github.com/okta-samples/okta-android-java-sample), and configures it to work with the OIDC app. This quickstart uses a basic Android project instead, as it's easier to understand the Okta-specific additions if you work through them yourself.
+> **Note**: If you're using the Okta CLI, you can also run `okta start android-kotlin` to create an app. This command creates an OIDC app in Okta, downloads the [okta-android-kotlin-sample](https://github.com/okta-samples/okta-android-kotlin-sample), and configures it to work with the OIDC app. This quickstart uses a basic Android project instead, as it's easier to understand the Okta-specific additions if you work through them yourself.
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/createproject.md:11]: I'm not sure if this note is really needed, though if it is...

```suggestion
> **Note**: The Okta CLI can create a Kotlin app for Android that's pre-configured for an Okta org by using the command `okta start android-kotlin`. This QuickStart uses a basic project to show you how to configure an app.
```

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/definecallback.md [hunk 2]
```diff
@@ -1,13 +1,15 @@
-Define the following placeholder inside your app's `build.gradle`:
+Parse and define the following placeholder inside your app's `build.gradle`:
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/definecallback.md:1]: I'm not sure that this new text adds anything. The developer isn't "Parsing" anything, they are adding content to the file that I assume is a placeholder. If that's the case, then the original text is fine. The only thing that may be worth adding is what the placeholder does.

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/getuserinfo.md [hunk 3]
```diff
@@ -1,42 +1,10 @@
-To find information about the user, use the `getUserProfile()` method. The `UserInfo` object passed into the `onSuccess()` method contains information for the signed-in user.
+To find information about the user, use the `idToken` method. The `Jwt` object contains information for the user.
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/getuserinfo.md:1]: This is missing the bridge between the shared text and the code snippets. In the iOS version of this I start with the What (Display the user info), then the how (by updating an empty function they've already entered). I then show the code. After the code I explain what it does.

Consider a similar flow. Lead with what the developer is adding to the app, printing the preferred username. Then have them add the code or perhaps explain how the Kotlin SDK deal with the ID Token. And after that, explain how it gets the local ID token from... etc.

The text that precedes the content of this file is:

> ## Get info about the user
> 
> The ID token returned by Okta contains user information, or *claims*, that are based on the scopes requested by the app (see [Configure your app](#configure-your-app)).
> 
> This app includes the `profile` scope that includes the user's email address, name, and preferred username. You can use this information to update your UI, such as showing the customer's name.
> 
> Use the Okta user information endpoints for items that aren't available in the ID token. For general information on requesting user info, see the [userinfo response example](/docs/reference/api/oidc/#response-example-success-6)).
> 
> <StackSnippet snippet="getuserinfo" />
>

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/opensignin.md [hunk 4]
```diff
@@ -1,173 +1,223 @@
 1. Update the default layout file for the app (found in `app/src/main/res/layout/`) to add a sign-in button, sign-out button, and a text view:
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/opensignin.md:1]: A slight order change... and what's the file?? There's just a path:
```suggestion
1. Add sign-in and sign-out buttons and a text view to the app's default layout file by adding the following code to `app/src/main/res/layout/`:
```

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/opensignin.md [hunk 5]
```diff
@@ -1,173 +1,223 @@
...
+            <ProgressBar
+                android:layout_gravity="center"
+                android:id="@+id/progress_bar"
+                android:layout_width="50dp"
+                android:layout_height="50dp"
+                />
+    </LinearLayout>
+    ```
+
+1. Fill in strings for the `text` values referenced above, in `app/src/main/res/values/strings.xml`:
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/opensignin.md:47]:
```suggestion
1. Add the following code to `app/src/main/res/values/strings.xml` to add the strings for the `text` values of the buttons:
```

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/opensignin.md [hunk 6]
```diff
@@ -1,173 +1,223 @@
...
+                    LoggedOut(errorMessage)
+                } else {
+                    LoggedIn.create(errorMessage)
+                }
+            }
+        }
+    }
+    ```
+
+1. Add the following content to your activity (for example `app/src/main/java/com/okta/android/samples/browser_sign_in/BrowserSignInActivity.kt`) to handle button clicks and browser state events:
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/opensignin.md:157]: Be prescriptive for filenames and classnames:
```suggestion
1. Add the following code Thant handles button clicks and browser state events to `app/src/main/java/com/okta/android/samples/browser_sign_in/BrowserSignInActivity.kt`:
```

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/opensignin.md [hunk 7]
```diff
@@ -1,173 +1,223 @@
...
+                    }
+                }
+            }
+        }
+    }
+    ```
+
+#### Running the sample
+
+To run the sample, see [the Android Studio documentation](https://developer.android.com/studio/run/device).
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/opensignin.md:223]: Is this needed? Same reason I asked before, in the iOS world developers are assumed to know Xcode and how to run on a device versus Simulator

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/testapp.md [hunk 8]
```diff
@@ -2,4 +2,4 @@
 1. [Configure an emulator](https://developer.android.com/studio/run/emulator) or [connect a physical device](https://developer.android.com/studio/run/device) in Android Studio
 2. Run the application using **Shift+F10** on Windows/Linux, **Control+R** on macOS, or click **Run** (the "Play" button) in the toolbar.
 
-If you followed along with this guide or cloned the [completed sample application](https://github.com/oktadev/okta-android-quickstart), the app with the sign-in button appears.
+If you followed along with this guide or cloned the [completed sample application](https://github.com/okta-samples/okta-android-kotlin-sample), the app with the sign-in button appears.
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/testapp.md:5]: This isn't really a helpful statement. We have to assume that they did follow the guide. Better to state the expected behavior so that a developer knows if something went wrong. I'd also leave out the reference to the sample... the only place it's relevant is the part that refers to samples... all it does here is add confusion. Describe what they see and what the flows are. Something like

> After the app opens, tap the Sign In button. The Okta sign-in browser opens over the app. After the user successfully signs in, the browser closes and the app ??does what, what's different, is there a Sign Out button ??

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/usetoken.md [hunk 9]
```diff
@@ -1,26 +1,24 @@
-To make authenticated requests to your resource server, get the access token from the `SessionClient`. The function below sends a request using `HttpURLConnection`:
+To make authenticated requests to your resource server, add the access token interceptor to your `OkHttpClient` instance.
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/usetoken.md:1]: This is partially repeating the general (see below). The focus of this content is the how. You don't need to say "to make authorized requests... etc. It should be how to make those requests (the iOS version is fairly simple as there's a convenience function that does everything for you.)

I think this may be code that creates that function for the developer. If so, then the text is something like:
Add the following code to <file>. It adds a convenience function that... etc.


> ## Use the access token
> 
> Your own server API (a resource server in OAuth 2.0) uses the Okta-generated access token for both authenticating that the user is signed in and ensuring that the user is authorized to perform the action or access the information.
> 
> Use the access token by adding it to the HTTP `Authorization` header of outgoing requests in your mobile or other client using this format:
> 
> ```
> Authorization: Bearer ${token}
> ```
> 
> Your API then checks incoming requests for valid tokens. To learn how to protect your API endpoints and require token authentication, see [Protect your API endpoints](/docs/guides/protect-your-api/).
> 
> <StackSnippet snippet="usetoken" />
>

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/usetoken.md [hunk 10]
```diff
@@ -1,26 +1,24 @@
...
+                // Authenticated successfully.
+            } else {
+                // Invalid status code.
+            }
+        }
+    })
 }
 ```
+
+The `accessTokenInterceptor` method attaches an `Authorization: Bearer ${token}` header with the access token to all requests. Credentials are internally refreshed when needed. If no valid access token is available, no authorization header will be added.
```
Comment (mauricesharp-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/usetoken.md:24]: I think that "Credentials are internally refreshed when needed" may mean that the SDK refreshes the token if needed... If so, then simpler language is better: "The SDK refreshes the token if that's required." or something like that.

The last two sentences are a double negative, which can be a bit hard to parse, how about:

> An authorization header is added only if there's a valid access token.

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/createproject.md [hunk 11]
```diff
@@ -1,18 +1,11 @@
-1. Create a new Java Android project with an **Empty Activity** using Android Studio (we called it `browser_sign_in`). For information on creating a project, see the [Android Studio documentation](https://developer.android.com/training/basics/firstapp/creating-project).
+1. Create a new Kotlin Android project with an **Empty Activity** using Android Studio (we called it `browser_sign_in`). For information on creating a project, see the [Android Studio documentation](https://developer.android.com/training/basics/firstapp/creating-project).
```
[the remark this answers is not in the document]
Comment (JayNewstrom) [reply packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/createproject.md:1]: This is mostly just a carry over from the previous guide. I like linking out to the android developer docs. I've gotten a lot of questions about how to do these things from people prototyping Android, that aren't Android experts. I don't have a strong feeling though here, so happy to remove it if you do.

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/createproject.md [hunk 12]
```diff
@@ -1,18 +1,11 @@
...
 
-2. Add the following permissions to your `AndroidManifest.xml`:
+2. The SDK uses features not available in all Android versions, add the desugaring SDK.
```
[the remark this answers is not in the document]
Comment (JayNewstrom) [reply packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/createproject.md:3]: The suggested text wasn't quite accurate, the next commit will contain an update (but please double check me again).

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/definecallback.md (hunk 2 above)
Comment (JayNewstrom) [reply packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/definecallback.md:1]: The code being added is parsing the manifest placeholder. I'm updating this section to make this more clear. Will be in the next commit.

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/addconfigpkg.md [hunk 13]
```diff
@@ -1,5 +1,9 @@
-Add the latest [Okta Android OIDC library](https://github.com/okta/okta-oidc-android) to the `dependencies` section of the `build.gradle` file:
+Add the latest [Okta Mobile Kotlin library](https://github.com/okta/okta-mobile-kotlin) to the `dependencies` block of the `app/build.gradle` file:
 
 ```gradle
-implementation 'com.okta.android:okta-oidc-android:1.2.2'
+// Ensure all dependencies are compatible using the Bill of Materials (BOM).
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/addconfigpkg.md:4]:
```suggestion
// Ensure that all dependencies are compatible using the Bill of Materials (BOM).
```

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/testapp.md [hunk 14]
```diff
@@ -1,5 +1,5 @@
 
 1. [Configure an emulator](https://developer.android.com/studio/run/emulator) or [connect a physical device](https://developer.android.com/studio/run/device) in Android Studio
-2. Run the application using **Shift+F10** on Windows/Linux, **Control+R** on macOS, or click **Run** (the "Play" button) in the toolbar.
+2. [Run the application](https://developer.android.com/studio/run/device) in Android Studio
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/testapp.md:3]:
```suggestion
2. [Run the application](https://developer.android.com/studio/run/device) in Android Studio.
```

File: packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/testapp.md [hunk 15]
```diff
@@ -1,5 +1,5 @@
...
 
-If you followed along with this guide or cloned the [completed sample application](https://github.com/oktadev/okta-android-quickstart), the app with the sign-in button appears.
+After the app opens, tap the Login button. The Okta sign-in browser opens over the app. After the user successfully signs in, the browser closes and the app displays a welcome message to the user, and has a Logout of Browser button.
```
Comment (susanharper-okta) [inline packages/@okta/vuepress-site/docs/guides/sign-into-mobile-app-redirect/main/android/testapp.md:5]:
```suggestion
After the app opens, tap the **Login** button. The Okta sign-in browser opens over the app. After the user successfully signs in, the browser closes and the app displays a welcome message to the user, and has a **Logout of Browser** button.
```

**Review by susanharper-okta — APPROVED** [review]
a couple nits. approving to not hold you up.
